src/functionaltests/java/com/ericsson/ei/utils/DataBaseManager.java

```diff
@@ -138,16 +134,19 @@ public boolean verifyAggregatedObjectExistsInDB() throws InterruptedException {
      * @return list of missing events
      */
     private List<String> compareSentEventsWithEventsInDB(List<String> checklist) {
-        mongoClient = new MongoClient(getMongoDbHost(), getMongoDbPort());
-        MongoDatabase db = mongoClient.getDatabase(database);
-        MongoCollection<Document> table = db.getCollection(collection);
-        List<Document> documents = table.find().into(new ArrayList<>());
-        for (Document document : documents) {
-            for (String expectedID : new ArrayList<>(checklist)) {
-                if (expectedID.equals(document.get("_id").toString())) {
-                    checklist.remove(expectedID);
+        MongoCollection<Document> table = getCollection(eventObjectCollectionName);
+        try {
+            List<Document> documents = table.find().into(new ArrayList<>());
+
+            for (Document document : documents) {
+                for (String expectedID : new ArrayList<>(checklist)) {
+                    if (expectedID.equals(document.get("_id").toString())) {
+                        checklist.remove(expectedID);
+                    }
```

Comment:
Should avoid streams since they are hard to read for newbs and slow.
This is old code and should not be seen as new in Github. 
Editing an incoming list is usually not good, but in this case it may be so, If anyone else feels different I may change. But would not really want to touch that code.

Reply:
I was only trying to relay how I would have written the code as I find it hard to read. 

On streams: If you feel they are hard to read I can respect your opinion, but as said in https://jaxenter.com/java-performance-tutorial-how-fast-are-the-java-8-streams-118830.html streams are not that slow and we don't write code for speed optimization.

Reply:
Steam speed depends on the number of elements in the list, where at 100k elements paralell steam is faster than normal for loop https://medium.com/@milan.mimica/slow-like-a-stream-fast-like-a-loop-524f70391182

Reply:
Did I get you right. You would like to rewrite it?

Reply:
Done.